lib/src/inline_parser.dart

```diff
@@ -274,6 +274,118 @@ class AutolinkSyntax extends InlineSyntax {
   }
 }
 
+/// Matches autolinks like `http://foo.com`.
+class AutolinkExtensionSyntax extends InlineSyntax {
+  static const START = r'(?:^|[\s*_~(>])';
+  static const SCHEME = r'(?:(?:https?|ftp):\/\/|www\.)';
+  static const DOMAIN_PART = r'\w\-';
+  static const DOMAIN = '[$DOMAIN_PART][$DOMAIN_PART.]+';
+  static const PATH = r'[^\s<]*';
+  // static const TRUNCATING_PUNCTUATION_NEG = r'[^\s<\?\!\.\,\:\*\_\~]';
+  static const TRUNCATING_PUNCTUATION_NEG = r'';
+
+  static const TRUNCATING_PUNCTUATION_POS = r'[?!.,:*_~]';
+
+  AutolinkExtensionSyntax()
+      : super('$START(($SCHEME)($DOMAIN)($PATH))$TRUNCATING_PUNCTUATION_NEG');
+
+  @override
+  bool tryMatch(InlineParser parser) {
+    var startMatch = pattern.matchAsPrefix(
+        parser.source, (parser.pos > 0 ? parser.pos - 1 : 0));
+    if (startMatch != null) {
+      // Write any existing plain text up to this point.
+      parser.writeText();
+
+      if (onMatch(parser, startMatch)) parser.consume(startMatch[0].length);
+      return true;
+    }
+
+    return false;
+  }
+
+  @override
+  bool onMatch(InlineParser parser, Match match) {
+    var url = match[1];
+    var href = url;
+    var matchLength = url.length;
+
+    if (url.startsWith(new RegExp(r'(\s|\>)'))) {
+      url = url.substring(1, url.length - 1);
+      href = href.substring(1, href.length - 1);
+      parser.pos++;
+      matchLength--;
+    }
+
+    /** Prevent accidental standard autolink matches */
+    if (url.endsWith('>') && parser.source[parser.pos - 1] == '<') {
+      return false;
+    }
+
+    /**
+     * When an autolink ends in ), we scan the entire autolink for the total
+     * number of parentheses. If there is a greater number of closing
+     * parentheses than opening ones, we don’t consider the last character
+     * part of the autolink, in order to facilitate including an autolink
+     * inside a parenthesis:
+     * https://github.github.com/gfm/#example-600
+     */
+    if (url.endsWith(')')) {
+      final opening = new RegExp(r'\(').allMatches(url).length;
```

Comment:
Let's not use RegExp just to count the number of parens. Can you write a private helper method that just walks the url and counts opening and closing? Should be much faster.

Reply:
Could I switch this to `url.allMatches('(').length;` for a similar speed improvement, or does this still use a RegExp underneath? The documentation doesn't seem to explicitly say that it does.